### PR TITLE
Avoid the use of registry.NewLogHasher.

### PR DESCRIPTION
**Pull request by pphaneuf**
Part of #2245.

### CODE REVIEW

File: client/log_verifier.go [hunk 1]
```diff
@@ -64,9 +64,8 @@ func NewLogVerifierFromTree(config *trillian.Tree) (*LogVerifier, error) {
 		return nil, fmt.Errorf("client: NewLogVerifierFromTree(): TreeType: %v, want %v or %v", got, log, pLog)
 	}
 
-	logHasher, err := registry.NewLogHasher(config.HashStrategy)
-	if err != nil {
-		return nil, fmt.Errorf("client: NewLogVerifierFromTree(): NewLogHasher(): %v", err)
+	if config.HashStrategy != trillian.HashStrategy_RFC6962_SHA256 {
+		return nil, fmt.Errorf("client: NewLogVerifierFromTree(): unknown hasher: %s", config.HashStrategy)
```
Comment (pavelkalinnikov) [inline client/log_verifier.go:68]:
```golang
if s := config.HashStrategy; s != trillian.HashStrategy_RFC6962_SHA256 {
	return nil, fmt.Errorf("client: NewLogVerifierFromTree(): unknown hasher: %v", s)
```
?
Same below.

File: server/admin/admin_server.go [hunk 2]
```diff
@@ -88,8 +87,8 @@ func (s *Server) CreateTree(ctx context.Context, req *trillian.CreateTreeRequest
 	}
 	switch tree.TreeType {
 	case trillian.TreeType_LOG, trillian.TreeType_PREORDERED_LOG:
-		if _, err := registry.NewLogHasher(tree.HashStrategy); err != nil {
-			return nil, status.Errorf(codes.InvalidArgument, "failed to create hasher for tree: %v", err.Error())
+		if tree.HashStrategy != trillian.HashStrategy_RFC6962_SHA256 {
+			return nil, status.Errorf(codes.InvalidArgument, "unknown hasher for tree: %s", tree.HashStrategy)
```
Comment (pavelkalinnikov) [inline server/admin/admin_server.go:91]: Drop " for tree"?

File: storage/mysql/log_storage.go [hunk 3]
```diff
@@ -228,13 +228,12 @@ func (m *mySQLLogStorage) beginInternal(ctx context.Context, tree *trillian.Tree
...
-	if err != nil {
-		return nil, err
+	if tree.HashStrategy != trillian.HashStrategy_RFC6962_SHA256 {
+		return nil, fmt.Errorf("unknown hashing strategy: %s", tree.HashStrategy)
 	}
 
-	stCache := cache.NewLogSubtreeCache(defaultLogStrata, hasher)
-	ttx, err := m.beginTreeTx(ctx, tree, hasher.Size(), stCache)
+	stCache := cache.NewLogSubtreeCache(defaultLogStrata, rfc6962.DefaultHasher)
+	ttx, err := m.beginTreeTx(ctx, tree, rfc6962.DefaultHasher.Size(), stCache)
```
[the remark this answers is not in the document]
Comment (pphaneuf) [reply storage/mysql/log_storage.go:236]: It seems to be used for some checking later on, not sure that it's really worthwhile, I agree. Not sure if I want to do this at all for now, but either way, I'd do it separately.

File: client/log_verifier.go (hunk 1 above)
Comment (pphaneuf) [reply client/log_verifier.go]: Done I also made the error messages a bit more consistent!

File: server/admin/admin_server.go (hunk 2 above)
Comment (pphaneuf) [reply server/admin/admin_server.go]: Done.

**Comment by pavelkalinnikov** [comment]
Please fix the test.

**Comment by pphaneuf** [comment]
I *think* it's just a flake? Re-running it, to be sure.

**Comment by pavelkalinnikov** [comment]
It seems related to this PR:
```
--- FAIL: TestServer_CreateTree (19.09s)
    --- FAIL: TestServer_CreateTree/invalidHashStrategy (0.00s)
        admin_server_test.go:509: CreateTree() = (_, "rpc error: code = InvalidArgument desc = unknown hash strategy: UNKNOWN_HASH_STRATEGY"), want (_, "unknown hasher")
```

**Comment by pphaneuf** [comment]
Ugh, I got fooled by irrelevant highlighting in GCB, thanks! Should be fixed now, I'll merge once it completes successfully.

**Pull request merged**